File: csharp/Spring.cs
using System;

namespace SmoothKit
{
    public static class Spring
    {
        public static float Update(
            float current,
            float target,
            ref float velocity,
            float tension = 20f,
            float damping = 0.8f,
            float dt = 1f / 60f)
        {
            float force = (target - current) * tension;
            velocity = (velocity + force * dt) * damping;
            current += velocity * dt;

            return current;
        }
    }
}

File: ts/spring.ts
export function spring(
    current: number,
    target: number,
    velocity: number,
    tension: number = 20,
    damping: number = 0.8,
    dt: number = 1 / 60
): { value: number; velocity: number } {
    const force = (target - current) * tension;
    velocity = (velocity + force * dt) * damping;
    current += velocity * dt;

    return { value: current, velocity };
}
